diff --git a/rules/base.xml.in b/rules/base.xml.in
index 2f64fa4..e0ee70f 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -1458,6 +1458,18 @@
             <_description>English (Workman, international with dead keys)</_description>
            </configItem>
          </variant>
+        <variant>
+          <configItem>
+            <name>viminsert</name>
+            <_description>English (VimInsert)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>vimnormal</name>
+            <_description>English (VimNormal)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
diff --git a/symbols/us b/symbols/us
index 833836d..7c584f3 100644
--- a/symbols/us
+++ b/symbols/us
@@ -475,8 +475,8 @@ xkb_symbols "dvorak-r" {
 };
 
 // Classic dvorak layout
-// by Piter Punk <piterpk@terra.com.br> - 2006-07-06 
-// Based on dvorak layout and e-mail from Russel L. Harris rlharris@oplink.net 
+// by Piter Punk <piterpk@terra.com.br> - 2006-07-06
+// Based on dvorak layout and e-mail from Russel L. Harris rlharris@oplink.net
 // on xorg list.
 partial alphanumeric_keys
 xkb_symbols "dvorak-classic" {
@@ -863,7 +863,7 @@ xkb_symbols "classmate-intl" {
 
     key <LSGT> { [ backslash,	bar,		backslash,	bar ] };
 
-    key <KP7>  { [	KP_Home,		KP_7,	
+    key <KP7>  { [	KP_Home,		KP_7,
 			onehalf,		dead_horn	] };
     key <KP8>  { [	KP_Up,			KP_8,
 			threequarters,		dead_ogonek	] };
@@ -942,7 +942,7 @@ xkb_symbols "classmate-altgr-intl" {
 
     key <LSGT> { [ backslash,	bar,		backslash,	bar ] };
 
-    key <KP7>  { [	KP_Home,		KP_7,	
+    key <KP7>  { [	KP_Home,		KP_7,
 			dead_horn,		dead_horn	] };
     key <KP8>  { [	KP_Up,			KP_8,
 			dead_ogonek,		dead_ogonek	] };
@@ -1278,6 +1278,31 @@ xkb_symbols "workman" {
 };
 
 partial alphanumeric_keys
+xkb_symbols "viminsert" {
+
+    include "us(basic)"
+    name[Group1]= "English (VimInsert)";
+
+    include "capslock(swapescape)"
+    include "altwin(ctrl_alt_win)"
+};
+
+partial alphanumeric_keys
+xkb_symbols "vimnormal" {
+
+    include "us(basic)"
+    name[Group1]= "English (VimNornal)";
+
+    include "capslock(swapescape)"
+    include "altwin(ctrl_alt_win)"
+
+    key <AC06> { [ Left, Left ] };
+    key <AC07> { [ Down, Down ] };
+    key <AC08> { [ Up, Up ] };
+    key <AC09> { [ Right, Right ] };
+};
+
+partial alphanumeric_keys
 xkb_symbols "workman-intl" {
 
     include "us(intl)"
